components/support/nimbus-fml/src/client/descriptor.rs

```diff
@@ -0,0 +1,91 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
```

Comment:
This code has been reviewed before; this moves it all into a file called `client/descriptor.rs`.